sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -316,20 +346,45 @@ class RocksDBFileManager(
     }
   }
 
+  // Get latest snapshot version <= version
+  def getLatestSnapshotVersionAndUniqueId(
+      version: Long, checkpointUniqueId: Option[String] = None): Array[(Long, Option[String])] = {
```

Comment:
This is from a similar way of how `getLatestSnapshotVersion` used in V1 is checked. When `fm.exists(path)` returns false, that function returns 0. This is because when first loading the path, the path does not exist. Here I use (0, None) to represent because version 0 does not have checkpoint id.

Note that because of the code refactoring suggested in the comments, the version = 0 part is specially handled in loadV2. But I feel as a utility function it is better to still consider this situation in case in the future people make any changes and are not aware of this fact.

Reply:
In the case of version = 0, it is clearer to me if the return value is Option[Array[String]]. If it is None, it means no snapshot is found. It is non-intuitive to say we found a verison 0, but it has no checkpoint ID.